src/Parser/Internal/AttributesDocBlockParser.php

```diff
@@ -0,0 +1,108 @@
+        $attributes = $this->reflection->getAttributes();
+        foreach ($attributes as $attribute) {
+            if ($attribute->getName() === self::COMMAND_ATTRIBUTE_CLASS_NAME) {
+                foreach ($attribute->getArguments() as $argName => $argValue) {
+                    switch ($argName) {
+                    case 'name':
+                        $this->commandInfo->setName($argValue);
+                        break;
+                    case 'description':
+                        $this->commandInfo->setDescription($argValue);
+                        break;
+                    case 'help':
+                        $this->commandInfo->setHelp($argValue);
+                        break;
+                    case 'aliases':
+                        $this->commandInfo->setAliases($argValue);
+                        break;
+                    case 'usage':
+                        $this->commandInfo->setExampleUsage(key($argValue), array_pop($argValue));
+                        break;
+                    case 'options':
+                        $set = $this->commandInfo->options();
+                        foreach ($argValue as $name => $option) {
+                            $variableName = $this->commandInfo->findMatchingOption($name);
+                            $description = trim(preg_replace('#[ \t\n\r]+#', ' ', $option['description']));
+                            list($description, $defaultValue) = $this->splitOutDefault($description);
+                            $set->add($variableName, $description);
+                            if ($defaultValue !== null) {
+                                $set->setDefaultValue($variableName, $defaultValue);
+                            }
+                        }
+                        break;
+                    case 'params':
+                        $set = $this->commandInfo->arguments();
+                        foreach ($argValue as $name => $param) {
+                            $variableName = $this->commandInfo->findMatchingOption($name);
+                            $description = trim(preg_replace('#[ \t\n\r]+#', ' ', $param['description']));
+                            list($description, $defaultValue) = $this->splitOutDefault($description);
+                            $set->add($variableName, $description);
+                            if ($defaultValue !== null) {
+                                $set->setDefaultValue($variableName, $defaultValue);
+                            }
+                        }
+                        break;
+                    default:
+                        foreach ($argValue as $name => $annotation) {
+                          foreach ($annotation as $value) {
+                            $this->commandInfo->addAnnotation($name, $value);
+                          }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    protected function splitOutDefault($description)
```

Comment:
These two methods are cut and paste from `BespokeDocBlockParser`. Happy to make a base class or trait these can share, but didn't want to refactor too much in this PR.

Reply:
We could reduce code duplication in a follow-on PR.

Reply:
Maybe just add a comment just in case.